dlt/common/schema/exceptions.py

```diff
@@ -156,3 +157,27 @@ def __init__(self, table_name: str) -> None:
 
 class ColumnNameConflictException(SchemaException):
     pass
+
+
+class UnboundColumnException(SchemaException):
+    def __init__(self, schema_name: str, table_name: str, column: TColumnSchemaBase) -> None:
+        self.column = column
+        self.schema_name = schema_name
+        self.table_name = table_name
+        nullable = column.get("nullable", True)
+        merge_key = column.get("merge_key", False)
+        primary_key = column.get("primary_key", False)
+        if merge_key:
+            col_type_msg = "merge key"
+        elif primary_key:
+            col_type_msg = "primary key"
+        elif nullable:
+            col_type_msg = "nullable"
+        else:
+            col_type_msg = "non-nullable"
+        super().__init__(
+            f"A {col_type_msg} column {column['name']} in table {table_name} in schema"
```

Comment:
I would make this error message better. Think about the perspective of the user that does not know the inner workings of dlt. It should be something like "The column xxx on table yyy is marked as non-nullable but did not receive any data during this load. This can happen if you manually specify non-nullable columns (for example primary or merge keys) in your code or import schema but this column does not match any incoming data."